features/eolearn/features/doubly_logistic_approximation.py

```diff
@@ -0,0 +1,84 @@
+"""
+Module for calculating doubly logistic approximation.
+"""
+import itertools as it
+import numpy as np
+from scipy.optimize import curve_fit
+from eolearn.core import EOTask
+
+
+def doubly_logistic(middle, initial_value, scale, a1, a2, a3, a4, a5):
+    # pylint: disable=invalid-name,locally-disabled
+    """
+    Function that is passed to scipy.optimize
+    """
+    return initial_value + scale * np.piecewise(middle, [middle < a1, middle >= a1],
+                                                [lambda y: np.exp(-((a1 - y) / a4) ** a5),
+                                                 lambda y: np.exp(-((y - a1) / a2) ** a3)])
+
+
+class DoublyLogisticApproximationTask(EOTask):
+    """
+    EOTask class for calculation of doubly logistic approximation on each pixel for a feature. The task creates new
+    feature with the function parameters for each pixel as vectors.
+    :param feature: A feature on which the function will be approximated
+    :type feature: str
+    :param new_feature: Name of the new feature where parameters of the function are saved
+    :type new_feature: str
+    :param initial_parameters: Initial parameter guess
+    :type initial_parameters: List of floats length 7 corresponding to each parameter
+    :param mask_feature: A feature used as a mask for valid regions. If left as None the whole patch is used
+    :type mask_feature: (FeatureType, str) or None
+    """
+
+    def __init__(self, feature, new_feature='DOUBLY_LOGISTIC_PARAM', initial_parameters=None, mask_feature=None):
+        self.feature = feature
+        self.initial_parameters = initial_parameters
+        self.new_feature = new_feature
+        self.mask_feature = mask_feature
+
+    def _fit_optimize(self, x_axis, y_axis):
+        """
+        :param x_axis: Horizontal coordinates of points
+        :type x_axis: List of floats
+        :param y_axis: Vertical coordinates of points
+        :type y_axis: List of floats
+        :return: List of optimized parameters [c1, c2, a1, a2, a3, a4, a5]
+        """
+        bounds_lower = [np.min(y_axis), -np.inf, x_axis[0], 0.15, 1, 0.15, 1, ]
+        bounds_upper = [np.max(y_axis), np.inf, x_axis[-1], np.inf, np.inf, np.inf, np.inf, ]
+        if self.initial_parameters is None:
+            self.initial_parameters = [np.mean(y_axis), 0.2, (x_axis[-1] - x_axis[0]) / 2, 0.15, 10, 0.15, 10]
+        optimal_values = curve_fit(doubly_logistic, x_axis, y_axis, self.initial_parameters,
```

Comment:
I think that the `sigma` parameter of the fit might be very useful as well, in cases where you have a mean and st.dev. NDVI and you want to fit the DL curve to these values, instead of the values of one pixel.

Reply:
Nevermind, I forgot this is done per pixel.